Feature/front search form

clementprdhomme opened the pull request:
NEW:
* Complete overhaul of the filters with custom accordion-like filters (with accessibility features), quick check/uncheck buttons, notification badges which display the number of selected options and a form verification feature to disable the apply button and display a warning message

CHANGE:
* Update the colors and boldness of the text and remove some texts according to the design changes

FIX:
* Issues with the general fonts: font-weight error and cross-browser rendering
* Issue with the vertical aligning of the font, fixed with a pseudo-element hack (font's fault for not centering correctly)

clementprdhomme commented:
@simaob or @rschumann could you take a look at the failing tests? They are related with the API.

simaob commented:
Some issues:
- Filtering Actors or Actions out doesn't seem to be working;
- If I selected something from the map and then do a search where that something doesn't show up, and then I click for more information it breaks the app and there's no "back to search" link. (e.g.: Select a micro actor, change search to ignore Micro actors, press apply. Boom)

clementprdhomme commented:
Hey @simaob,

* I just added the missing code for filtering the actors and actions, optimizing the requests so we won't fetch a collection we don't need.
* I fixed the issue which stuck the UI when going through your steps. Even if the back buttons appears, my commit doesn't prevent the main issue which is letting the app goes into an inconsistent state. I've already planned to review the whole interaction with the map in the to-dos.

The PR is ready to merge once you review it.
